Initialize storybook

fcollonval opened the pull request:
This initialize the storybook with a first button component:

![Screenshot 2023-12-21 at 17-51-20 Components _ Button - Docs ⋅ Storybook](https://github.com/datalayer/jupyter-ui/assets/8435071/a0dcdbc1-049a-4a55-88f7-388a3ac24eb1)

- It is possible to switch between dark and light theme (dynamic loading of the CSS properties).
- Option to compare with Lab button or not
- Some text about what is requires for the lab button to be styled properly
- Change primary button style to use Lab brand color (aka blueish)

---

I tried to change the danger button to always have a red background (not only on hover). But it is unfortunately not easy because the background is hard coded with the default button background. A solution would be to override the sx props, but do we really want to go that road.

echarles reviewed (approved):
LGTM. Thx @fcollonval

echarles commented:
> I tried to change the danger button to always have a red background (not only on hover). But it is unfortunately not easy because the background is hard coded with the default button background. A solution would be to override the sx props, but do we really want to go that road.

They have the same color, which is already very good.

echarles commented on packages/react/package.json:
```diff
@@ -128,7 +128,7 @@
     "@jupyterlab/ui-components-extension": "4.0.3",
     "@lumino/default-theme": "2.1.2",
-    "@primer/react": "0.0.0-20230912151940",
+    "@primer/react": "36.4.0",
```
@fcollonval  This introduces a failure Attempted import error: 'UnderlineNav' is not exported from '@primer/react/drafts' (imported as 'UnderlineNav').

I faced that some time ago. primer has a policy, maybe possible to use something more stable than 0.0.0-..., but for now, keeping 0.0.0-20230912151940 would be ok

fcollonval replied:
In which package did you see the failure - the components is not draft anymore so the import should be updated to `@primer/react`

echarles replied:
I have seen that failure here https://github.com/datalayer/jupyter-ui/actions/runs/7291010284/job/19869056711#step:7:1471

echarles commented:
> . A solution would be to override the sx props, but do we really want to go that road.

We could have JupyteLabButton components in the main source (like the Cell, Notebook... components) that wraps a Primer button, uses the jupyter theme and overrides where needed the color, border...

echarles commented:
As additional components, we can have specific components and examples for

- Text
- Title
- Input

We could also have example for icons pulled from https://github.com/datalayer/icons

- Icon

fcollonval commented:
@echarles this is ready for a second look

I 
- added notes for text and headings
- improved the style for the button
- added a skeleton for icons; the lab icon does not work for now as webpack is unable to load the SVG file from the lab package in the storybook :unamused: 
- reverted the react primer version

echarles commented:
> added a skeleton for icons; the lab icon does not work for now as webpack is unable to load the SVG file from the lab package in the storybook 😒

I faced that issue some time ago in another storybook and solved it using a cusom webpack.config.js as described on https://storybook.js.org/docs/builders/webpack

This is what is was looking like (probably not working anymore, but can give ideas)

```
const path = require("path");

const SRC_PATH = path.join(__dirname, './../src');
const STORIES_PATH = path.join(__dirname, './../stories');

// dont need stories path if you have your stories inside your components folder
module.exports = ({config}) => {

  // https://github.com/webpack-contrib/css-loader/issues/295#issuecomment-644131089
  config.module.rules = config.module.rules.filter(rule => rule.test.toString().indexOf('css') == -1)
  
  config.module.rules.push({
    // This is so that we can compile any React,
    // ES6 and above into normal ES5 syntax.
    test: /\.(js|jsx)$/,
    use: [
      {
      loader: "babel-loader",
      options: {
        presets: [
          "@babel/preset-react",
        ],
        cacheDirectory: true
      }
    }],
  });

  config.module.rules.push({
    test: /\.(ts|tsx)$/,
    include: [
      SRC_PATH, 
      STORIES_PATH
    ],
      use: [
        {
          loader: require.resolve('awesome-typescript-loader'),
          options: {
            configFileName: './tsconfig.json'
          }
        },
//        https://github.com/storybookjs/storybook/issues/3968        
        { loader: require.resolve('react-docgen-typescript-loader') }
      ],
    });
  config.resolve.extensions.push('.ts', '.tsx', 'js', 'jsx');

  config.module.rules.push({
    test: /\.css?$/,
    use: ['style-loader', 'css-loader'],
  });
  config.resolve.extensions.push('css');

  // Remove rule for images defined by file-loader
  config.module.rules = config.module.rules.filter(
    rule => ! rule.test.toString().includes('jpg')
  );

  config.module.rules.push({
    test: /\.(png|jpg|jpeg|gif|ttf|woff|woff2|eot)(\?v=[0-9]\.[0-9]\.[0-9])?$/,
    use: [{ loader: 'url-loader', options: { limit: 10000 } }],
  });
  config.resolve.extensions.push('png', 'jpg', 'jpeg', 'gif', 'ttf', 'woff', 'woff2', 'eot');

  config.module.rules.push({
    test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,
    oneOf: [
      {
        // in css files, svg is loaded as a url formatted string
        issuer: /\.css$/ ,
        use: {
          loader: 'svg-url-loader',
          options: { encoding: 'none', limit: 10000 }
        }
      },
      {
        // in js, jsx, ts, and tsx files svg is loaded as a raw string
        issuer: /\.(js|jsx|ts|tsx)$/,
        use: {
          loader: 'raw-loader'
        }
      }
    ],
  });
  config.resolve.extensions.push('svg');

  config.resolve.alias =  Object.assign(config.resolve.alias, { 
    "crypto": "crypto-browserify",
    "path": "path-browserify",
    "stream": "stream-browserify",
    "vm": "vm-browserify"
    "assets": path.resolve(__dirname, './../src/assets/'),
  });

//  const babelLoader = config.module.rules[0];
//  babelLoader.exclude.push(require.resolve('./node_modules'));

  return config;

};
```

This can be added a subsequent PR.

echarles reviewed (approved):
LGTM Thx @fcollonval

fcollonval commented:
Thanks for the tip @echarles 
I think the webpack rule to add is the one that inline SVG within J's files.
